$NetBSD$

Preserve SCO host selection and the NetBSD atomic feature check.
Let genccode write assembly for every NetBSD target, not only i386.

--- configure.orig
+++ configure
@@ -5833,6 +5833,7 @@
 *-dec-osf*) icu_cv_host_frag=mh-alpha-osf ;;
 *-*-nto*)	icu_cv_host_frag=mh-qnx ;;
 *-ncr-*)	icu_cv_host_frag=mh-mpras ;;
+*-*-sco3.2v5*)	icu_cv_host_frag=mh-scoosr5 ;;
 *-wasi*)	icu_cv_host_frag=mh-wasi ;;
 *) 		icu_cv_host_frag=mh-unknown ;;
 esac
@@ -6461,6 +6462,29 @@
 fi
 
 
+# check if sys/atomic.h is present.
+for ac_header in sys/atomic.h
+do :
+  ac_fn_c_check_header_mongrel "$LINENO" "sys/atomic.h" "ac_cv_header_sys_atomic_h" "$ac_includes_default"
+if test "x$ac_cv_header_sys_atomic_h" = xyes; then :
+  cat >>confdefs.h <<_ACEOF
+#define HAVE_SYS_ATOMIC_H 1
+_ACEOF
+
+fi
+
+done
+
+case "${host}" in
+*-netbsd*)
+if test "x$ac_cv_header_sys_atomic_h" = "xyes"; then
+    CONFIG_CPPFLAGS="$CONFIG_CPPFLAGS -DU_HAVE_NETBSD_ATOMIC_OPS=1";
+fi
+;;
+*)
+;;
+esac
+
 U_ENABLE_DYLOAD=1
 enable=yes
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether to enable dynamic loading of plugins. Ignored if plugins disabled." >&5
@@ -7009,7 +7033,7 @@
 # Check to see if genccode can generate simple assembly.
 GENCCODE_ASSEMBLY=
 case "${host}" in
-*-linux*|*-kfreebsd*-gnu*|i*86-*-*bsd*|*-pc-gnu)
+*-linux*|*-netbsd*|*-kfreebsd*-gnu*|i*86-*-*bsd*|*-pc-gnu)
     if test "$GCC" = yes; then
         # We're using gcc, and the simple -a gcc command line works for genccode
         GENCCODE_ASSEMBLY="-a gcc"
